systems/plants/test/compareParsersmex.cpp

```diff
@@ -89,18 +89,18 @@ void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) {
 
 		  auto matlab_H = matlab_model->massMatrix<double>();
 		  auto cpp_H = cpp_model->massMatrix<double>();
-		  valuecheck(matlab_H.value(),cpp_H.value(),1e-8,"H doesn't match");
+		  valuecheckMatrix(matlab_H.value(),cpp_H.value(),1e-8,"H doesn't match");
```

Comment:
There was a conflict with
```
template <typename T>
void valuecheck(const T& a, const T& b)
```
that only appeared in GCC. I believe this is a bug in GCC. Renaming this function is a workaround for that. This took me quite a while to figure out. I documented it extensively in testUtil.h:
```
/*
 * +++tk:
 * Note: can't call this function 'valuecheck' because it conflicts with our valuecheck template for general types above on GCC.
 * (yes, even though the number of template arguments and the number of function arguments is different!)
 * It does compile on MSVC 2010 and Clang.
 * What I believe happens is that during argument-dependent name lookup, GCC instantiates Eigen::MatrixBase<int>, which causes errors deep in Eigen::internal code
 * Possibly related to this discussion:
 * https://stackoverflow.com/questions/25925551/gcc-and-clang-implicitly-instantiate-template-arguments-during-operator-overload
 *
 * Fairly minimal failure example (still involving Eigen):
 * #include <Eigen/Core>
 * #include <iostream>
 *
 * template <typename Derived>
 * void foo(const Eigen::MatrixBase<Derived>& m, double bla) {
 *   std::cout << m << ", " << bla << std::endl;
 * }
 *
 * template <typename T>
 * void foo(const T& t) {
 *   std::cout << t << std::endl;
 * }
 *
 * int main() {
 *   int i = 1;
 *   foo<int>(i); // doesn't compile
 *   foo(i); // compiles
 *   return 0;
 * }
 */
template<typename DerivedA, typename DerivedB>
void valuecheckMatrix(const Eigen::MatrixBase<DerivedA>& a, const Eigen::MatrixBase<DerivedB>& b, double tol, std::string error_msg = "")
```